packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -169,6 +175,37 @@ export class QueuedRequestController extends BaseController<
       `${controllerName}:enqueueRequest`,
       this.enqueueRequest.bind(this),
     );
+
+    this.messagingSystem.subscribe(
+      SelectedNetworkControllerEventTypes.stateChange,
+      (_, patch) => {
```

Comment:
Nit: Have you considered caching connected domains here instead of inspecting the patch? e.g.

```
#connectedDomains = Set<string>;

...

this.messagingSystem.subscribe(
  SelectedNetworkControllerEventTypes.stateChange,
  (currentConnectedDomains) => {
    const removedDomains = this.#connectedDomains.difference(currentConnectedDomains);
    
    for (const removedDomain of removedDomains) {
       this.#flushQueueForOrigin(removedDomain);
    }
    
    this.#connectedDomains = currentConnectedDomains;
  },
  (state) => new SetObject.keys(state.domains)),
}
```

This would be more resilient as it would continue to work even if different types of patch operations were used, or if multiple domains were removed at once.

Reply:
Wait sorry, re-thinking this, I was misinterpreting the code here. I'll delete this and re-read....